Mathlib/NumberTheory/EllipticDivisibilitySequence.lean

```diff
@@ -102,6 +105,42 @@ lemma IsDivSequence.smul (h : IsDivSequence W) (x : R) : IsDivSequence (x • W)
 lemma IsEllDivSequence.smul (h : IsEllDivSequence W) (x : R) : IsEllDivSequence (x • W) :=
   ⟨h.left.smul x, h.right.smul x⟩
 
+end IsEllDivSequence
+
+/-- Strong recursion principle for a normalised EDS: if we have
+ * `P 0`, `P 1`, `P 2`, `P 3`, and `P 4`,
+ * for all `m : ℕ` we can prove `P (2 * (m + 3))` from `P k` for all `k < 2 * (m + 3)`, and
+ * for all `m : ℕ` we can prove `P (2 * (m + 2) + 1)` from `P k` for all `k < 2 * (m + 2) + 1`,
+then we have `P n` for all `n : ℕ`. -/
+@[elab_as_elim]
+noncomputable def normEDSRec' {P : ℕ → Sort u}
```

Comment:
Can you provide a bit of doc explaining the diff between these two `def`s and why this is primed/

Reply:
The only difference is N vs Z as above, which is already mentioned in the docstring.

Again, I can't use `ofNat` because it's already the name of a lemma below.

Reply:
OK, but then please come up with a name so that `Int` and `Nat` show up.